parcels/tools/timer.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import print_function
+import time
+
+
+class Timer():
+    def __init__(self, name, parent=None, start=True):
+        self._start = -1
+        self._end = -1
+        self._t = 0
+        self._name = name
+        self._children = []
+        self._parent = parent
+        if self._parent:
+            self._parent._children.append(self)
+        if start:
+            self.start()
+
+    def start(self):
+        if self._parent:
+            assert(self._parent._start > 0), ("Timer '%s' cannot be started. Its parent timer does not run" % self._name)
+        self._start = time.time()
+
+    def stop(self):
+        assert(self._start > 0), ("Timer '%s' was stopped before being started" % self._name)
+        self._t += time.time() - self._start
+        self._start = -1
+
+    def print_local(self):
+        if self._start > 0:
+            print("Timer '%s': %g s (process running)" % (self._name, self._t + time.time() - self._start))
+        else:
+            print("Timer '%s': %g s" % (self._name, self._t))
+
+    def local_time(self):
+        if self._start > 0:
+            return self._t + time.time() - self._start
+        else:
+            return self._t
+
+    def print_tree(self, step=0, root_time=0, parent_time=0):
+        time = self.local_time()
+        if step == 0:
+            root_time = time
+        print(('(%3d%%)' % round(time/root_time*100)), end='')
+        for i in range(step+1):
+            print('  ', end='')
+        if step > 0:
+            print('(%3d%%) ' % round(time/parent_time*100), end='')
+        print("Timer '%s': %1.1e s" % (self._name, time))
```

Comment:
Can we use `timedelta` objects throughout for prettier printing here? I.e. not everything in seconds, but also in minutes, hours where appropriate?